File: go/1512-number-of-good-pairs.go
func numIdenticalPairs(nums []int) int {
    pairs := 0

    for i := 0; i < len(nums); i++ {
        for j := i+1; j < len(nums); j++ {
            if nums[i] == nums[j] {
                pairs++
            }
        }
    }

    return pairs
}

File: cpp/1512-number-of-good-pairs.cpp
class Solution { 
        unordered_map<int, int> Memo = {};    
        int numIdenticalPairs(vector<int> & nums){            
            unordered_map<int, int> Memo;            
            int i, k;            
            int NGood;            
            NGood = 0;            
            Memo = {};            
            for(int & i : nums){                
                if(Memo.find(i) == Memo.end()){                    
                    Memo.insert(make_pair(i, 1));                
                }
                else{                    
                    NGood = NGood + Memo[i];                    
                    Memo[i]++;                    
                }                
            }            
            return NGood;            
        }    
};
